ocaml/networkd/lib/network_config.ml

```diff
@@ -56,9 +56,20 @@ let read_management_conf () =
     in
     debug "Firstboot file management.conf has: %s"
       (String.concat "; " (List.map (fun (k, v) -> k ^ "=" ^ v) args)) ;
-    let device = List.assoc "LABEL" args in
-    let vlan =
-      if List.mem_assoc "VLAN" args then Some (List.assoc "VLAN" args) else None
+    let vlan = List.assoc_opt "VLAN" args in
+    let bond_mode =
+      Option.value ~default:"" (List.assoc_opt "BOND_MODE" args)
+    in
+    let bond_members =
+      String.split_on_char ',' (List.assoc "BOND_MEMBERS" args)
+    in
+    let device =
+      (* Take 1st member of bond *)
+      match (bond_mode, bond_members) with
+      | "", _ | _, [] ->
+          List.assoc "LABEL" args
```

Comment:
The original code already used this method:
```ocaml
let device = List.assoc "LABEL" args in
```

I guess it's because there always is a `LABEL` in the `management.conf`. Would prefer use `assoc_opt`? Which default value should I use then? Or which error should I throw?

Reply:
I think using `List.assoc_opt` makes it easier to throw a meaningful exception. If the label is coming from a file, it's strictly not an internal error but a user error, but if this was never handled before, an internal error is still better than what we had before.

Reply:
On it

Reply:
I think it's already done [here](https://github.com/xapi-project/xen-api/blob/master/ocaml/networkd/lib/network_config.ml#L167-L171), do you prefere a specific treatment here?

Reply:
If this is caused by missing something in a config file, it should say so - so yes.

Reply:
Would something like this work?
```ocaml
    let device =
      (* Take 1st member of bond *)
      match (bond_mode, bond_members) with
      | "", _ | _, [] ->
          match (List.assoc_opt "LABEL" args) with
          | Some x -> x
          | None ->
              error "No `LABEL` in `management.conf`" ;
              raise Read_error
      | _, hd :: _ ->
          hd
    in
 ```

Reply:
```
error "%s: missing LABEL in %s" __FUNCTION__ management_conf
```